frontend/src/pages/Compare.tsx

```diff
@@ -14,56 +14,115 @@
  * limitations under the License.
  */
 
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { useQuery } from 'react-query';
 import { ApiRunDetail } from 'src/apis/run';
 import { QUERY_PARAMS } from 'src/components/Router';
 import { FeatureKey, isFeatureEnabled } from 'src/features';
 import { Apis } from 'src/lib/Apis';
 import { errorToMessage } from 'src/lib/Utils';
 import { URLParser } from '../lib/URLParser';
-import CompareV1, { CompareState } from './CompareV1';
+import CompareV1 from './CompareV1';
 import CompareV2 from './CompareV2';
 import { PageProps } from './Page';
 
-export type CompareProps = PageProps & CompareState;
+enum CompareVersion {
+  V1,
+  V2,
+  Mixed,
+  InvalidRunCount,
+  Unknown,
```

Comment:
I added an `Unknown` state so that the page would not incorrectly show a banner or different page upon initial load (see the `compareVersion` state variable). However, if there is a better way to go about this, please let me know.

Reply:
There is a `isPending` return value from `useQuery`:

`const { isPending, isError, error } = useQuery`

So when the page is loading data from backend, you can check this condition `isPending === True` to tell whether it is currently loading.

Reply:
You are right, the right term is `isLoading` in this case. I agree that we should avoid showing misleading banner even for a period of a blink. If the `Unknown` state is helpful in overcoming this issue, let's go for it. Thank you Chris!